src/Microsoft.Extensions.ProjectModel.Sources/ProjectContextExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.ProjectModel
+{
+    internal static class ProjectContextExtensions
```

Comment:
These extensions seem wasteful. 

Reasons: 
- First is trying to avoid using ignorecase. Our dev guidelines state to always specify comparison.
- Second is overly specific to user secrets. Let the projects who care about user secrets add this API; no need to pollute the IntelliSense list with extra info when it typically will not be needed.